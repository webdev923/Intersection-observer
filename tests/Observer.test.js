import React from 'react'
import { shallow, mount } from 'enzyme'
import Observer from '../src/index.js'

jest.mock('../src/intersection')

it('Should render <Observer />', () => {
  const callback = jest.fn()
  shallow(<Observer>{callback}</Observer>)
  expect(callback).toHaveBeenCalledWith(false)
})

it('Should render <Observer /> inview', () => {
  const callback = jest.fn()
  const wrapper = shallow(<Observer>{callback}</Observer>)
  wrapper.setState({ inView: true })
  expect(callback).toHaveBeenLastCalledWith(true)
})

it('Should render <Observer /> with children outside view', () => {
  const wrapper = shallow(
    <Observer className="observer">
      <div>Content</div>
    </Observer>,
  )
  expect(wrapper).toMatchSnapshot()
})

it('Should render <Observer /> with children inview', () => {
  const wrapper = shallow(
    <Observer className="observer">
      <div>Content</div>
    </Observer>,
  )
  expect(wrapper).toMatchSnapshot()
})

it('Should not render <Observer /> render outside view', () => {
  const wrapper = shallow(
    <Observer
      className="observer"
      render={inView => <div>Inview: {inView.toString()}</div>}
    />,
  )
  expect(wrapper).toMatchSnapshot()
})

it('Should render <Observer /> render when in view', () => {
  const wrapper = shallow(
    <Observer
      className="observer"
      render={inView => <div>Inview: {inView.toString()}</div>}
    />,
  )
  wrapper.setState({ inView: true })

  expect(wrapper).toMatchSnapshot()
})

it('Should get innerRef', () => {
  const ref = jest.fn()
  const wrapper = mount(<Observer className="observer" innerRef={ref} />)
  const node = wrapper.getDOMNode()

  expect(ref).toHaveBeenCalledWith(node)
})
