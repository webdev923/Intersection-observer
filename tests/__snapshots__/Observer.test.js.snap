// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should not render <Observer /> render outside view 1`] = `
<div
  className="observer"
>
  <div>
    Inview: 
    false
  </div>
</div>
`;

exports[`Should render <Observer /> render when in view 1`] = `
<div
  className="observer"
>
  <div>
    Inview: 
    true
  </div>
</div>
`;

exports[`Should render <Observer /> with children inview 1`] = `
<div
  className="observer"
>
  <div>
    Content
  </div>
</div>
`;

exports[`Should render <Observer /> with children outside view 1`] = `
<div
  className="observer"
>
  <div>
    Content
  </div>
</div>
`;
